### PR TITLE
feat(IdentityCenter): add identity center custom policy

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add identity center custom policy
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add identity center custom policy
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
go test .\huaweicloud\services\acceptance\identitycenter\ -v -run TestAccCustomPolicyAttachment_basic
=== RUN   TestAccCustomPolicyAttachment_basic
=== PAUSE TestAccCustomPolicyAttachment_basic
=== CONT  TestAccCustomPolicyAttachment_basic
--- PASS: TestAccCustomPolicyAttachment_basic (11.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    12.416s
```

### CODE REVIEW

File: docs/incubating/identitycenter_custom_policy_attachment.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+subcategory: "IAM Identity Center"
+---
+
+# huaweicloud_identitycenter_custom_policy_attachment
+
+Manages an Identity Center custom policy attachment resource within HuaweiCloud.
```
Comment (Jason-Zhang9309) [inline docs/incubating/identitycenter_custom_policy_attachment.md:7]: add a note that only one resource can be created

File: docs/incubating/identitycenter_custom_policy_attachment.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `custom_policy` - (Required, String) Specifies the custom policy to attach to a permission set.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID which is formatted `<instance_id>/<permission_set_id>`.
```
Comment (Jason-Zhang9309) [inline docs/incubating/identitycenter_custom_policy_attachment.md]: the format is permission_set_id in the code

File: huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_custom_policy_attachment.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+			"custom_policy": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the custom policy to attach to a permission set.`,
+			},
+		},
+	}
+}
+
+func resourceCustomPolicyAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_custom_policy_attachment.go]: you can merge this func with resourceCustomPolicyAttachmentUpdate

Comment (houpeng80) [reply huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_custom_policy_attachment.go]: done

File: docs/incubating/identitycenter_custom_policy_attachment.md (hunk 2 above)
Comment (houpeng80) [reply docs/incubating/identitycenter_custom_policy_attachment.md]: fixed

File: docs/incubating/identitycenter_custom_policy_attachment.md (hunk 1 above)
Comment (houpeng80) [reply docs/incubating/identitycenter_custom_policy_attachment.md:7]: done

**Pull request merged**